python-avd/pyavd/_eos_designs/shared_utils/wan.py

```diff
@@ -214,17 +214,14 @@ def get_public_ip_for_wan_interface(
             EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3InterfacesItem
             | EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3PortChannelsItem
         ),
-    ) -> str:
+    ) -> str | None:
         """
-        If not a WAN route-server this returns public IP and if not found then the interface IP without a mask.
+        If not a WAN route-server this returns public IP and if not found then the interface IP without a mask or None if no ip is set.
 
         For WAN route-servers we try to find the IP under wan_route_servers.path_groups.interfaces.
         If not found we look for the public_ip and then the ip_address under the interface.
         If there is no public_ip and if ip_address is "dhcp" we raise an error.
         """
-        if not self.is_wan_server:
-            return default(interface.public_ip, get_ip_from_ip_prefix(interface.ip_address))
-
```

Comment:
Removing this means we will run through a lot of extra logic for every non wan server device, just to fall through to the end.

Reply:
yes but we need the extra logic:

if there is a public_ip we want it
if there is no ip_address or it is dhcp we don't want to call get_ip_from_prefix

and otherwise we want get_ip_from_ip_prefix


the only difference is taht for a wan_server we should raise if we have no public IP or if it is dhcp whereas for a wan interface, we don't need to for dhcp and even if we are missing the ip_address (though we would raise later as per the test if there is no ip_address at all)